src/client/components/PreferencesDialog.vue

```diff
@@ -42,52 +30,25 @@ export default Vue.extend({
         target.classList.add('language-' + this.lang);
       }
     },
-    updatePreferencesFromStorage(): Map<
-            string,
-            boolean | string
-            > {
-      for (const k of preferences) {
-        const val = PreferencesManager.load(k);
-        if (k === 'lang') {
-          PreferencesManager.preferencesValues.set(k, this.$data[k]);
-          this[k] = val || 'en';
-          PreferencesManager.preferencesValues.set(k, val || 'en');
-        } else {
-          const boolVal = val !== '' ? val === '1' : this.$data[k];
-          PreferencesManager.preferencesValues.set(k, val === '1');
-          this.$data[k] = boolVal;
-        }
-      }
-      return PreferencesManager.preferencesValues;
-    },
-    updatePreferences(): void {
-      let strVal: string = '';
-      for (const k of preferences) {
-        const val = PreferencesManager.preferencesValues.get(k);
-        if (val !== this.$data[k]) {
-          if (k === 'lang') {
-            strVal = this.$data[k];
-          } else {
-            strVal = this.$data[k] ? '1' : '0';
-          }
-          PreferencesManager.save(k, strVal);
-          PreferencesManager.preferencesValues.set(k, this.$data[k]);
-          this.setPreferencesCSS(this.$data[k], k);
-        }
+    updatePreferences() {
+      for (const k of Object.keys(this.preferencesManager.values) as Array<Preference>) {
+        const val = this[k];
```

Comment:
```suggestion
        const val = this.preferencesManager.values[k];
```

As written this is assuming that `this.data` is set to `this.preferencesManager.values()`.

Reply:
That's right. If something changes this, well, the whole thing would need to change a little bit.